core/tls-websocket-server/src/stream_state.rs

```diff
@@ -35,9 +36,9 @@ pub(crate) type RustlsWebSocket = WebSocket<RustlsStream>;
 /// Internal TLS stream state. From pure TLS stream, to web-socket handshake and established WS.
 pub(crate) enum StreamState {
 	Invalid,
-	TlsStream(RustlsStream),
+	TlsStream(Box<RustlsStream>),
 	WebSocketHandshake(RustlsMidHandshake),
-	EstablishedWebsocket(RustlsWebSocket),
+	EstablishedWebsocket(Box<RustlsWebSocket>),
```

Comment:
Was this one also because of a big discrepancy between enum variant sizes?

Reply:
Yes, I will add it to the PR description as well.